### DISCUSSION TITLE
Using the useImperativeHandle with a dynamically imported component

### DISCUSSION THREAD

**Discussion by sirmyron** [vercel/next.js#27517] [Help]
My setup includes a `Parent` component that dynamically importing a `Table` component and a `Chart` component depending on the format selected by the user in the **Parent** component. I'm dynamically importing them to make sure the heavy chart library gets properly split.

In my **Table** and **Chart** components I've exposed an `export` function using the `useImperativeHandle` hook, where each implements the necessary logic to do an export. However, when I try to access the function using a ref to the rendered component `console.log(renderedComponentRef.current)`, all I see is an object with a `retry` function and the `export` function is never available. I only got this to work after removing the `dynamic` import. This happens with any component I dynamically load (from my code or from an external library).

I'm somewhat new to React so I'm thinking there's something I don't fully understand here, or perhaps this isn't working as expected. 

Could anyone shed some light on this for me please?

**Comment by Ehesp** [comment]
See this issue: https://github.com/vercel/next.js/issues/4957
